### PR TITLE
Removal of Nodev6

### PR DESCRIPTION
**Pull request by Jagatjot**
Fixes Issue #346

### CODE REVIEW

**Comment by Jagatjot** [comment]
@fgalan travis tests giving same issue of AMQP timeout. Could you please re-run the tests and review the PR? Thanks!

**Comment by fgalan** [comment]
Thank you very very much for your contribution. Could you add the following line to CHANGES_NEXT_RELEASE about the change in this PR, pls?

> Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions)

**Comment by Jagatjot** [comment]
> Thank you very very much for your contribution. Could you add the following line to CHANGES_NEXT_RELEASE about the change in this PR, pls?
> 
> > Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions)

Fixed by https://github.com/telefonicaid/iotagent-json/pull/400/commits/b6a2286b9101f280f5e57f219a090dbcbdd679a0

**Comment by Jagatjot** [comment]
> LGTM. Thanks!
> 
> PR is ready although we are going to delay the merge after we can release first a new version of the IOTA (after that, maybe a conflict on CHANGES_NEXT_RELEASE would need to be solved, although it will be trivial).
> 
> On the meanwhile, what about this similar issues related with v6 removal? Do you plan also to address them? It will be great :)
> 
> * [telefonicaid/iotagent-node-lib#700](https://github.com/telefonicaid/iotagent-node-lib/issues/700)
> * [telefonicaid/iotagent-ul#304](https://github.com/telefonicaid/iotagent-ul/issues/304)
> * [telefonicaid/lwm2m-node-lib#138](https://github.com/telefonicaid/lwm2m-node-lib/issues/138)
> * [telefonicaid/lightweightm2m-iotagent#140](https://github.com/telefonicaid/lightweightm2m-iotagent/issues/140)

Yes, I will be addressing them too. Thank you!

File: .travis.yml [hunk 1]
```diff
@@ -15,20 +15,6 @@ install:
 
 jobs:
   include:
-    - stage: test
```
Comment (jason-fox) [inline .travis.yml]: Given that Node 12 was release on 2019-04-23 and is now current - this should be updated to test against Node 12 rather than deleted

Comment (fgalan) [reply .travis.yml]: I'd suggest to add a a block for node 12 just after the one for node 10 and see what happens.

Btw @jason-fox do you know which node 12 hasn't set yet as current in https://github.com/nodejs/Release table? (I'm looking rigth now and I see "Pending").

Comment (Jagatjot) [reply .travis.yml]: Fixed by https://github.com/telefonicaid/iotagent-json/pull/400/commits/1f31e8c7d6c6939a3ec1d042a34752711de46599

Comment (jason-fox) [reply .travis.yml]: @fgalan - according to https://travis-ci.org/telefonicaid/iotagent-node-lib/jobs/526754611 Travis is using `12.1.0`

```console
nvm install 12
```
```text
Downloading and installing node v12.1.0...
Downloading https://nodejs.org/dist/v12.1.0/node-v12.1.0-linux-x64.tar.xz...
Computing checksum with sha256sum
Checksums matched!
Now using node v12.1.0 (npm v6.9.0)
```

Comment (fgalan) [reply .travis.yml]: > Btw @jason-fox do you know which node 12 hasn't set yet as current in https://github.com/nodejs/Release table? (I'm looking rigth now and I see "Pending").

Now it reads "Current release". It seems they changed 4 days ago (https://github.com/nodejs/Release/pull/440/files).

**Comment by fgalan** [comment]
> PR is ready although we are going to delay the merge after we can release first a new version of the IOTA (after that, maybe a conflict on CHANGES_NEXT_RELEASE would need to be solved, although it will be trivial).

That version (1.10.0) has been just released so this PR would be ready once CHANGES_NEXT_RELEASE conflict gets solved, as far as I understand. @Jagatjot could you manage it, please?

**Comment by fgalan** [comment]
@Jagatjot any update in this PR, please? Don't hesitate of asking if something is not clear about what is pending in this PR to be finished and merged. Thanks!

**Comment by Jagatjot** [comment]
Fixed by https://github.com/telefonicaid/iotagent-json/pull/400/commits/84c99628e1b840ee57bcce57a9e76f24b4c80f0c

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Fix: modified engine version >=6 in packages.json
+Fix: update logops dependency from 1.0.8 to 2.1.0
+Add: config.http.timeout (and associated enviroment variable IOTA_HTTP_TIMEOUT)(#152)
+Add: config.mqtt.{cert,key,protocol,rejectUnauthorized} (and associated environment variables IOTA_MQTT_*)(#372)
+Add: readding sinon as dev dependency (~6.1.0)
+Upgrade NodeJS version from 8.12.0 to 8.15.0 in Dockerfile due to security issues
+Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: I'm afraid the conflict in CHANGES_NEXT_RELEASE has not been correctly solved. Note that at the present moment that file is empty in master (see https://github.com/telefonicaid/iotagent-json/blob/master/CHANGES_NEXT_RELEASE). Thus, the changes in this file in this PR are only the new things added by this PR.

In sum, this file should contain only one line:

```
Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions) 
```

Comment (Jagatjot) [reply CHANGES_NEXT_RELEASE]: Fixed by https://github.com/telefonicaid/iotagent-json/pull/400/commits/def895a461a0e867c8f965dc1f96da884c23efcc

**Comment by Chandradeep-NEC** [comment]
@fgalan 
Hi, Please close the issue as the PR has been merged successfully.

**Pull request merged**